### ISSUE TITLE
Make logging message key 'event' configurable

### ISSUE THREAD

**Issue by AdrianLC** [hynek/structlog#35]
Hello,

Nice project. Really loving it :)

Our product works with objects that represent touristic events and so I would like to use the
`'event'` key for the serialization of these objects. But the `'event'`  key is currently reserved for the logging message. Would you agree to make [these](https://github.com/hynek/structlog/blob/master/structlog/_base.py#L133) hard-coded key configurable? i.e. `structlog.configure(event_message_key="message")`

Seems pretty easy to implement. I could work on a pull request if you like the idea.

Thank you!

**Comment by hynek** [comment]
Hi I’m sorry but that IMHO add too much of complexity that would fall on our feet eventually. :(  I suggest you just use `event_` or simply `e` if it’s that common in your source base?

**Comment by AdrianLC** [comment]
Well... Doesn't seem complex at all to me. We could do it with a simple `@property` with setter and, of course, it would be an optional setting. Your choice though...

**Comment by hynek** [comment]
Wouldn’t it be easier for you to do that in a custom logger?  It’s a really cornery corner case and all the docs and examples rely on talking about events and event_dicts.  I really don’t want to add any more cognitive burden (TBH, I find the docs already too difficult to read due to structlog’s configurability; but again that configurability empowers you to swap out the logger for your own).

**Comment by cyrilgdn** [comment]
Is it still not configurable?

Applications using structlog are the only ones using event as log field in our stack (and almost the only ones where it's not configurable)

So we have to put the same custom processor in all our apps using structlog  in order to rename this field (it makes me sad :( )

**Comment by hynek** [comment]
While it’s regrettable that my software makes someone sad, it’s indeed not.  It’s nothing a lot of people asked for in over four years and it’s nothing I care about personally.

That said, if someone could show me a PR that is not too invasive and doesn’t have a performance impact on everyone else, I’d consider merging it.

**Comment by cyrilgdn** [comment]
Thanks for your quick answer, structlog is awesome anyway :) ! 

To be not too invasive, do you think it could be possible to have directly in structlog a processor to rename this field ?
Something like:

```

class EventRenamer(object):  # with a better name hopefully
    def __init__(self, field_name):
        self._field_name = field_name

    def __call__(self, logger, name, event_dict):
        if 'event' in event_dict and not isinstance(event_dict['event'], dict):
            event_dict[field_name] = event_dict.pop('event')
        return event_dict

```

If yes, I'll do a PR with pleasure!

**Comment by hynek** [comment]
I don’t think that’s OP was asking for?  He wanted to use the event key himself.  And yes, it seems to be just a matter of a simple processor which is why I’m rather reluctant to change anything in structlog proper.

**Comment by mmindenhall** [comment]
FWIW, I'd also be interested in this feature, but for a different reason.  I don't want to use the `event` key myself, but I'm working in an event-driven microservice architecture, and when we talk about "events", we're referring to application events, not log events.  So I'd prefer to rename this to `msg` or `message` to avoid confusion.

**Comment by ajcolaizzi-decentrix** [comment]
> FWIW, I'd also be interested in this feature, but for a different reason. I don't want to use the `event` key myself, but I'm working in an event-driven microservice architecture, and when we talk about "events", we're referring to application events, not log events. So I'd prefer to rename this to `msg` or `message` to avoid confusion.

I second this. Mostly because our other teams use "message" not "event" so we would like to remain consistent. 

I have really enjoyed structlog so far and it has added so much value for us!

**Comment by hynek** [comment]
Y'all, for those who want to rename `event` in the _output_, you just need these three lines:

```python
def rename_event_key(_, __, ed):
    ed["message"] = ed.pop("event")
    return ed
```

OP wanted something much more intrusive: use the `event` key for something else. E.g. `log.info("foo", event="bar")` which would require a lot more work and open heart surgery in `structlog`'s innards.

---

It's not super pretty but the original problem could be solved using something like this:

```python
def rename_event_key(_, __, ed):
    ed["message"] = ed.pop("event")
    ed["event"] = ed.pop("event_")
    return ed
```

And then you do `log.info("foo", event_="bar")` – using a trailing underscore to avoid naming clashes is a well-established pattern in Python.

Again, if that's not acceptable, writing an own logger is a matter of a dozen lines too.

---

LMK if I'm missing something or something is unclear.

**Comment by houmie** [comment]
Sorry where do I have to paste these three lines:

```
def rename_event_key(_, __, ed):
    ed["message"] = ed.pop("event")
    return ed
```

Do I have to extend any class? Not quite clear to me.
The word event in log output is problematic for filebeat and ELK stack, since it's a reserved word. It would be ideal if I could change it in the output to something else.

**Comment by houmie** [comment]
Sorry @hynek where do I have to paste these three lines, please:

```
def rename_event_key(_, __, ed):
    ed["message"] = ed.pop("event")
    return ed
```

Do I have to extend any class? Not quite clear to me.
The word event in log output is problematic for filebeat and ELK stack, since it's a reserved word. It would be ideal if I could change it in the output to something else.

Thanks

**Comment by hynek** [comment]
That’s a processor that you add to your processor chain. See https://www.structlog.org/en/stable/processors.html for details on processors.

**Comment by hynek** [comment]
Just for the record: only 7 years later, the just released 22.1 comes with [`structlog.processors.EventRenamer`](https://www.structlog.org/en/stable/api.html#structlog.processors.EventRenamer) that allows you to rename `event` to whatever you want, and whatever else you want to `event`.
